spec/lib/agris/client_spec.rb

```diff
@@ -33,11 +33,33 @@
 
   context 'when the post is processed' do
     let(:fixture_file) { 'create_order_processed.xml' }
+    context 'with the default dataset' do
+      it 'should return a result with status of Processed' do
+        result = client.create_order(agris_new_order)
 
-    it 'should return a result with status of Processed' do
-      result = client.create_order(agris_new_order)
+        expect(result.status).to eq 'Processed'
+      end
+    end
 
-      expect(result.status).to eq 'Processed'
+    context 'with a new  dataset' do
+      let!(:client) do
+        options = {
+          logger: Logger.new(STDOUT),
+          request_type: Agris::SavonRequest
+        }
+        Agris::Client.new(
+          context,
+          Agris::Credentials::Anonymous.new,
+          '007',
+          options
+        )
+      end
+
+      it 'should return a result with status of Processed' do
+        result = client.create_order(agris_new_order)
+
+        expect(result.status).to eq 'Processed'
```

Comment:
You aren't actually looking at what dataset is used so these tests don't tell us if your changes work or not.

The dataset is in the XML generated. Existing tests do this comparison via fixtures. So you should make a brand new fixture with a 007 dataset and make sure it matches

https://github.com/westernmilling/agris.rb/blob/master/spec/lib/agris/client_spec.rb#L65